sdks/python/apache_beam/io/gcp/datastore/v1new/types_test.py

```diff
@@ -134,6 +135,36 @@ def testQuery(self):
 
     logging.info('query: %s', q)  # Test __repr__()
 
+  def testRuntimeFilters(self):
+    filter_list = [
+        [(StaticValueProvider(str, 'property_name'),  # Filter 1
+          StaticValueProvider(str, '='),
+          StaticValueProvider(str, 'value'))],
+        [(StaticValueProvider(str, 'property_name'),  # Filter 2(1)
+          StaticValueProvider(str, '='),
+          StaticValueProvider(str, 'value')),
+         ('property_name', '=', 'value')],  # Filter 2(2)
+    ]
+    for filters in filter_list:
+      projection = ['f1', 'f2']
+      order = projection
+      distinct_on = projection
+      ancestor_key = Key(['kind', 'id'], project=self._PROJECT)
+      q = Query(kind='kind', project=self._PROJECT, namespace=self._NAMESPACE,
+                ancestor=ancestor_key, filters=filters, projection=projection,
+                order=order, distinct_on=distinct_on)
+      cq = q._to_client_query(self._test_client)
```

Comment:
This is testing `_set_runtime_filters` with itself.
Assuming all filter tuples have the same expected values, this test can be written and simplified as:
```py
filter_list = [...]
expected_filter = ('property_name', '=', 'value')
for filters in filter_list:
  ...
  q = ...
  cq = ...
  for _filter in cq.filters:
    self.assertEqual(expected_filter, _filter)
```
(the other asserts are covered in `testQuery`, but you can keep them if you want)

Reply:
Thanks for the feedback! I've updated the test to test:
`[(ValueProvider)] == [(str-filter)]`

and a composite filter of:
`[(ValueProvider), (str-filter)] == [(str-filter), (str-filter)]`